src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -117,32 +128,38 @@ def get_all_updates(self, cached=False):
             self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(cached), len(self.all_updates_cached)))
             return self.all_updates_cached, self.all_update_versions_cached  # allows for high performance reuse in areas of the code explicitly aware of the cache
 
-        out = self.invoke_package_manager(self.tdnf_check)
+        out = self.invoke_package_manager(self.__generate_command(self.tdnf_check, self.max_patch_publish_date))
         self.all_updates_cached, self.all_update_versions_cached = self.extract_packages_and_versions(out)
-
         self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(False), len(self.all_updates_cached)))
         return self.all_updates_cached, self.all_update_versions_cached
 
     def get_security_updates(self):
-        """Get missing security updates. NOTE: Classification based categorization of patches is not available in Azure Linux as of now"""
-        self.composite_logger.log_verbose("[TDNF] Discovering 'security' packages...")
-        security_packages, security_package_versions = [], []
+        """Get missing security updates. NOTE: Classification based categorization of patches is not available in TDNF as of now"""
+        self.composite_logger.log_verbose("[TDNF] Discovering all packages as 'security' packages, since TDNF does not support package classification...")
+        security_packages, security_package_versions = self.get_all_updates(cached=False)
         self.composite_logger.log_debug("[TDNF] Discovered 'security' packages. [Count={0}]".format(len(security_packages)))
         return security_packages, security_package_versions
```

Comment:
This is my mistake from earlier high-level discussions - the correct CentOS alignment is this -- https://github.com/Azure/LinuxPatchExtension/pull/318/files#r2241037265

The effective outcome is sort of the same but not quite. Make note of both scenarios - default patching, and customer-initiated and how they differ.

Reply:
The linked CentOS alignment does exist for default patching within TDNF as well. IIRC, our past discussion and conclusion to fetch from get_all_updates() within security, was to ensure these patches are tagged as 'Security' in status, which is needed to mark them all as Security updates for the customer